app/com/arpnetworking/pillar/Configuration.java

```diff
@@ -0,0 +1,92 @@
+/**
+ * Copyright 2017 Smartsheet.com
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.arpnetworking.pillar;
+
+import com.google.common.collect.Maps;
+import com.typesafe.config.Config;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import javax.inject.Inject;
+
+/**
+ * Configuration for the Pillar evolustions.
+ *
+ * @author Brandon Arp (brandon dot arp at smartseet dot com)
+ */
+public class Configuration {
+    /**
+     * Public constructor.
+     *
+     * @param configuration Play configuration
+     */
+    @Inject
+    public Configuration(final play.Configuration configuration) {
+        _datastores = Maps.newHashMap();
+
+        final Config dbConfig = configuration.underlying().getConfig("cassandra.db");
+        final Set<String> dbNames = dbConfig.root().keySet();
+        for (final String dbName : dbNames) {
+            _datastores.put(dbName, new DatastoreConfig(dbName, dbConfig.getConfig(dbName)));
+        }
+    }
+
+    public Map<String, DatastoreConfig> getDatastores() {
+        return _datastores;
+    }
+
+    private final Map<String, DatastoreConfig> _datastores;
+
+    /**
+     * Represents configuration for a single Cassandra datastore.
+     *
+     * @author Brandon Arp (brandon dot arp at smartseet dot com)
+     */
+    public static final class DatastoreConfig {
+        public String getDirectory() {
+            return _directory;
+        }
+
+        public List<String> getHosts() {
+            return _hosts;
+        }
+
+        public String getKeyspace() {
+            return _keyspace;
+        }
+
+        public Map<String, Object> getReplication() {
+            return _replication;
+        }
+
+        private DatastoreConfig(final String dbName, final Config config) {
+            if (config.hasPath("migration.directory")) {
+                _directory = config.getString("migration.directory");
+            } else {
+                _directory = "cassandra/migration/" + dbName;
+            }
+            _hosts = config.getStringList("hosts");
+            _keyspace = config.getString("keyspace");
+            _replication = config.getObject("replication").unwrapped();
+        }
+
+        private final String _directory;
```

Comment:
This is actually resolved through the environment as a resource.  I don't think it actually makes sense as a path because of it.